riscv/src/runtime.rs

```diff
@@ -194,31 +194,31 @@ impl Runtime {
 
     pub fn with_poseidon(mut self) -> Self {
         self.add_submachine(
-            "std::machines::hash::poseidon_gl::PoseidonGL",
+            "std::machines::hash::poseidon_gl_memory::PoseidonGLMemory",
             None,
-            "poseidon_gl",
-            [format!(
-                "instr poseidon_gl link ~> {};",
-                instr_link("poseidon_gl.poseidon_permutation", 12, 4)
-            )],
-            12,
+            "poseidon_gl(memory, split_gl)",
+            [r#"instr poseidon_gl X, Y
+                    link ~> val1_col = regs.mload(X, STEP)
+                    link ~> val2_col = regs.mload(Y, STEP + 1)
+                    link ~> poseidon_gl.poseidon_permutation(val1_col, val2_col, STEP)
+                {
+                    // make sure val1_col and val2_col are aligned memory addresses
+                    val3_col * 4 = val1_col,
+                    val4_col * 4 = val2_col,
+                    // make sure the remainders fit in 32 bits
+                    val3_col = X_b1 + X_b2 * 0x100 + X_b3 * 0x10000 + X_b4 * 0x1000000,
+                    val4_col = Y_b5 + Y_b6 * 0x100 + Y_b7 * 0x10000 + Y_b8 * 0x1000000
```

Comment:
So in the end this verifies that each address is a 34-Bit number that is divisible by 4, right? I think that's sufficient, we have other constraints to make sure the address is not above 32 Bit.